app/questionnaire/questionnaire_schema.py

```diff
@@ -130,6 +130,42 @@ def get_questions_for_block(block_json):
         for question_json in block_json.get('questions', []):
             yield question_json
 
+    @staticmethod
+    def is_summary_section(section):
+        """Summary section is a section with a single Summary group
+        """
+        if section['groups']:
+            return QuestionnaireSchema.is_summary_group(section['groups'][0])
+
+        return False
+
+    @staticmethod
+    def is_summary_group(group):
+        """Summary group is a group with a single block of type "Summary"
+        """
+        if group['blocks']:
+            return group['blocks'][0]['type'] == 'Summary'
```

Comment:
Same 2 comments as above. we shouldn't have a group with no blocks and we don't stop summary being in a group with other blocks so we should check all the blocks